core-plugins/src/main/java/io/cdap/plugin/transform/LogParserTransform.java

```diff
@@ -98,21 +99,27 @@ public LogParserTransform(LogParserConfig config) {
   @Override
   public void configurePipeline(PipelineConfigurer pipelineConfigurer) throws IllegalArgumentException {
     super.configurePipeline(pipelineConfigurer);
+    // Get failure collector for updated validation API
+    FailureCollector collector = pipelineConfigurer.getStageConfigurer().getFailureCollector();
     if (!S3_LOG.equals(config.logFormat) && !CLF_LOG.equals(config.logFormat) &&
       !CLOUDFRONT_LOG.equals(config.logFormat)) {
       LOG.error("Log format not currently supported.");
-      throw new IllegalStateException("Unsupported log format: " + config.logFormat);
+      collector.addFailure(String.format("Unsupported log format: '%s'", config.logFormat),
+          "Log format must be one of the following: S3, CLF, or CLOUDFRONT.").withConfigProperty("logFormat");
     }
     Schema inputSchema = pipelineConfigurer.getStageConfigurer().getInputSchema();
     if (inputSchema != null) {
       if (!inputSchema.getType().equals(Schema.Type.RECORD)) {
-        throw new IllegalArgumentException("Only Input Schema of type Schema.Type.RECORD is supported");
+        collector.addFailure("Only Input Schema of type Schema.Type.RECORD is supported",
```

Comment:
error : `Input schema must be of type record.` and keep corrective action null

Reply:
Fixed